src/transform.ts

```diff
@@ -97,6 +98,57 @@ export interface AggregatedFieldDef {
   as: string;
 }
 
+export interface WindowFieldDef {
+  /**
+   * The operations supported for the window aggregation
+   */
+  op: AggregateOp | WindowOnlyOp;
+
+  /**
+   * The optional parameters for the operation
+   */
+  param?: Number;
+
+  /**
+   * The field that will be used in the operation, some operations
+   * do not require fields
+   */
+  field?: string;
+
+  /**
+   * The name for the new field in the window transform
```

Comment:
full stop.. can you add it all for the rest of the docs.

Reply:
Resolved.